cabal-install/Distribution/Client/CmdInstall.hs

```diff
@@ -358,6 +434,20 @@ symlinkBuiltExe verbosity sourceDir destDir exe = do
     exe
     $ unUnqualComponentName exe
 
+-- | Create 'GhcEnvironmentFileEntry's for packages with exposed libraries.
+entriesForLibraryComponents :: TargetsMap -> [GhcEnvironmentFileEntry]
+entriesForLibraryComponents = Map.foldrWithKey' (\k v -> mappend (go k v)) []
+  where
+    hasLib :: (ComponentTarget, [TargetSelector]) -> Bool
+    hasLib (ComponentTarget CLibName _,        _) = True
+    hasLib (ComponentTarget (CSubLibName _) _, _) = True
+    hasLib _                                      = False
+    
+    go :: UnitId -> [(ComponentTarget, [TargetSelector])] -> [GhcEnvironmentFileEntry]
+    go unitId targets
+      | any hasLib targets = [GhcEnvFilePackageId unitId]
```

Comment:
```❯ cat ~/.ghc/x86_64-darwin-8.4.2/environments/default
package-id lnr-1.20.7-359f8114```

That's what I get after I try to install `linear`. Why do I not get the vowels?

Reply:
Are you on mac os?

https://github.com/typedrat/cabal/blob/638c78e790721566780dbf75a2e6f884ceafa970/cabal-install/Distribution/Client/PackageHash.hs#L164

Reply:
Hmm, how would I get around that...

Reply:
Not sure that removing the vowels _guarantees_ any shortening of the name, or contributes big savings.  
In any case, it is quite alienating to look into `.cabal/store` and see that the Grinch stole all the vowels.
I am feeling like I am trolled here, or there was some jester in the team developing the new cabal, and that jester is trying to make fun of the Haskell programmers.

Reply:
I filed #7209.